### PR TITLE
Add deprecation warnings for get_units_map and get_angular_units_map

### PR DESCRIPTION
**Pull request by snowman2**
- [x] Related to units changes in #599 (https://github.com/OSGeo/PROJ/pull/2065)
 - [x] Tests updated

### CODE REVIEW

File: pyproj/_list.pyx [hunk 1]
```diff
@@ -74,6 +75,12 @@ def get_units_map():
     dict:
         Units supported by PROJ
     """
+    warnings.warn(
+        "The behavior of 'pyproj.get_units_map' is deprecated "
+        "and will change in version 3.0.0.",
```
Comment (jorisvandenbossche) [inline pyproj/_list.pyx:80]: Small comment: it's not very clear what the consequence of this is: what will change? What should the user do instead when seeing this warning?

Comment (snowman2) [reply pyproj/_list.pyx:80]: The user really cannot do much here as the change will just happen. The exact change isn't 100% decided, but will definitely be different.

Comment (jorisvandenbossche) [reply pyproj/_list.pyx:80]: Then it is maybe a bit too soon to raise a warning? As this will just be confusing, I might think

Comment (snowman2) [reply pyproj/_list.pyx:80]: Well, I wanted to have something. Not sure if there are going to be any more 2.x releases after this next one.

**Pull request merged**